force_torque_sensor_broadcaster/src/force_torque_sensor_broadcaster.cpp

```diff
@@ -0,0 +1,175 @@
+    node->declare_parameter<std::string>("interface_names.force.z", "");
+    node->declare_parameter<std::string>("interface_names.torque.x", "");
+    node->declare_parameter<std::string>("interface_names.torque.y", "");
+    node->declare_parameter<std::string>("interface_names.torque.z", "");
+    node->declare_parameter<std::string>("frame_id", "");
+  } catch (const std::exception & e) {
+    fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
+    return controller_interface::return_type::ERROR;
+  }
+
+  return controller_interface::return_type::OK;
+}
+
+CallbackReturn ForceTorqueSensorBroadcaster::on_configure(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  sensor_name_ = node_->get_parameter("sensor_name").as_string();
+  interface_names_.resize(6, "");
+  interface_names_[0] = node_->get_parameter("interface_names.force.x").as_string();
+  interface_names_[1] = node_->get_parameter("interface_names.force.y").as_string();
+  interface_names_[2] = node_->get_parameter("interface_names.force.z").as_string();
+  interface_names_[3] = node_->get_parameter("interface_names.torque.x").as_string();
+  interface_names_[4] = node_->get_parameter("interface_names.torque.y").as_string();
+  interface_names_[5] = node_->get_parameter("interface_names.torque.z").as_string();
+
+  if (sensor_name_.empty() &&
+    std::count(interface_names_.begin(), interface_names_.end(), "") == 6)
+  {
+    RCLCPP_ERROR(
+      get_node()->get_logger(), "'sensor_name' or at least one "
+      "'interface_names.[force|torque].[x|y|z]' parameter has to be specified.");
+    return CallbackReturn::ERROR;
+  }
+
+  if (!sensor_name_.empty() &&
+    std::count(interface_names_.begin(), interface_names_.end(), "") != 6)
+  {
+    RCLCPP_ERROR(
+      get_node()->get_logger(), "both 'sensor_name' and "
+      "'interface_names.[force|torque].[x|y|z]' parameter can not be specified together.");
+    return CallbackReturn::ERROR;
+  }
+
+  frame_id_ = node_->get_parameter("frame_id").as_string();
+  if (frame_id_.empty()) {
+    RCLCPP_ERROR(get_node()->get_logger(), "'frame_id' parameter has to be provided.");
+    return CallbackReturn::ERROR;
+  }
+
+  if (!sensor_name_.empty()) {
+    force_torque_sensor_ = std::make_unique<semantic_components::ForceTorqueSensor>(
+      semantic_components::ForceTorqueSensor(sensor_name_));
+  } else {
+    force_torque_sensor_ = std::make_unique<semantic_components::ForceTorqueSensor>(
+      semantic_components::ForceTorqueSensor(
+        interface_names_[0], interface_names_[1],
+        interface_names_[2], interface_names_[3], interface_names_[4], interface_names_[5]));
+  }
+
+  try {
+    // register ft sensor data publisher
+    sensor_state_publisher_ = node_->create_publisher<geometry_msgs::msg::WrenchStamped>(
+      "~/sensor_states", rclcpp::SystemDefaultsQoS());
+    realtime_publisher_ = std::make_unique<StatePublisher>(sensor_state_publisher_);
+  } catch (...) {
```

Comment:
is there a class of exceptions that we could capture here? it'd be nice to have an error print, otherwise poor old soul whose broadcaster fails here won't know what's happening

Reply:
Added below exception in this place.

```
  } catch (const std::exception & e) {
    fprintf(stderr, "Exception thrown during publisher creation at configure stage with message
      : %s \n", e.what());
    return CallbackReturn::ERROR;
  }
```